tests/test_bookmarks.py

```diff
@@ -387,30 +379,27 @@ def bookmarks_test(self):
 
                         sync_record = sync_records[0]
 
-                        if stream == 'payments':
-                            self.assertPaymentsEqual(updated_record, sync_record)
-                        elif stream == 'inventories':
-                            self.assertInventoriesEqual(updated_record, sync_record)
-                        else:
-                            self.assertDictEqual(updated_record, sync_record)
+                        self.assertRecordsEqual(stream, updated_record, sync_record)
 
-    def assertInventoriesEqual(self, expected_record, sync_record):
-        self.assertEqual(frozenset(expected_record.keys()), frozenset(sync_record.keys()), "Expected keys in expected_record to equal keys in sync_record. [expected_record={}][sync_record={}]".format(expected_record, sync_record))
-        expected_record_copy = deepcopy(expected_record)
-        sync_record_copy = deepcopy(sync_record)
+    def assertRecordsEqual(self, stream, expected_record, sync_record):
+        if stream == 'payments':
+            self.assertDictEqualWithOffKeys(expected_record, sync_record, {'updated_at'})
+        elif stream == 'inventories':
+            self.assertDictEqualWithOffKeys(expected_record, sync_record, {'calculated_at'})
+        elif stream in {'employees', 'roles'}:
+            self.assertDictEqualWithOffKeys(expected_record, sync_record, {'created_at', 'updated_at'})
+        else:
+            self.assertDictEqual(expected_record, sync_record)
 
-        # Square api workflow updates these values so they're a few seconds different between the time the record is created and the tap syncs, but other fields are the same
-        self.assertGreaterEqual(sync_record_copy.pop('calculated_at'),
-                                expected_record_copy.pop('calculated_at'))
-        self.assertDictEqual(expected_record_copy, sync_record_copy)
-
-    def assertPaymentsEqual(self, expected_record, sync_record):
+    def assertDictEqualWithOffKeys(self, expected_record, sync_record, off_keys=set()):
```

Comment:
Can we add a comment explaining this method is only valid for full table streams and that it is a workaround to square? I am just worried this pattern will get picked up on a different tap.

Reply:
Oh  I guess we already say this below.